rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -1,98 +1,107 @@
-// generated from rosidl_generator_cpp/resource/msg__struct.hpp.em
-// generated code does not contain a copyright notice
-
-@#######################################################################
-@# EmPy template for generating <msg>__struct.hpp files
-@#
-@# Context:
-@#  - spec (rosidl_parser.MessageSpecification)
-@#    Parsed specification of the .msg file
-@#  - subfolder (string)
-@#    The subfolder / subnamespace of the message
-@#    Either 'msg' or 'srv'
-@#  - get_header_filename_from_msg_name (function)
-@#######################################################################
-@
-@{
-header_guard_parts = [
-    spec.base_type.pkg_name, subfolder,
-    get_header_filename_from_msg_name(spec.base_type.type) + '__struct_hpp']
-header_guard_variable = '__'.join([x.upper() for x in header_guard_parts]) + '_'
-}@
-#ifndef @(header_guard_variable)
-#define @(header_guard_variable)
-
+@# Included from rosidl_generator_cpp/resource/idl__struct.hpp.em
 // Protect against ERROR being predefined on Windows, in case somebody defines a
 // constant by that name.
-#if defined(_WIN32) && defined(ERROR)
-  #undef ERROR
+#if defined(_WIN32)
+  #if defined(ERROR)
```

Comment:
Here's the relevant commit 3d7cb50

There is a comment at L25 explaining `ERROR` potentially clashes with user-defined constants (the comment could be updated to also mention `NO_ERROR`). Specifically, `NO_ERROR` is defined in [tf2_msgs/TF2Error](http://docs.ros.org/api/tf2_msgs/html/msg/TF2Error.html).

Reply:
Since the comment is already present (https://github.com/ros2/rosidl/pull/334/files#diff-8c3e8f645f194b7db43c438d3896ad3eR2) I will mark this as resolved.